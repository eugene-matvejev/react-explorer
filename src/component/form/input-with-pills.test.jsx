import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import InputWithPills from './input-with-pills';

configure({ adapter: new Adapter() });

describe('<InputWithPills/>', () => {
    const props = {
        value: [
            {
                label: '{{label}}',
            }
        ]
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<InputWithPills {...props} />);

            expect(c).toMatchSnapshot();
        });
    });

    describe('with optional props', () => {
        [
            ['data-cy', '{{data-cy}}'],
        ].forEach(([prop, v]) => {
            it(`[::${prop}] as "${v}"`, () => {
                const c = shallow(<InputWithPills {...props} {...{ [prop]: v }} />);

                expect(c).toMatchSnapshot();
            });
        });
    });
});


