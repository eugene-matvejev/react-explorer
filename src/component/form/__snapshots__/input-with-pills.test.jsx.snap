// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputWithPills/> render with default/required props 1`] = `
<div
  className="input-with-pills"
>
  <Pill
    data-cy="-pill-0"
    key="0"
    label="{{label}}"
  />
  <HTMLInput
    className="input-with-pills__input"
    data-cy=""
  />
</div>
`;

exports[`<InputWithPills/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<div
  className="input-with-pills"
>
  <Pill
    data-cy="{{data-cy}}-pill-0"
    key="0"
    label="{{label}}"
  />
  <HTMLInput
    className="input-with-pills__input"
    data-cy="{{data-cy}}"
  />
</div>
`;

exports[`<InputWithPills/> render with optional props [::onClick] as "() => {}" 1`] = `
<div
  className="input-with-pills"
  onClick={[Function]}
>
  <Pill
    data-cy="-pill-0"
    data-id={0}
    key="0"
    label="{{label}}"
  />
  <HTMLInput
    className="input-with-pills__input"
    data-cy=""
  />
</div>
`;
